src/GitVersionTask.MsBuild/LibGit2Sharp/GitLoaderContext.cs

```diff
@@ -11,22 +10,28 @@ namespace GitVersion.MSBuildTask.LibGit2Sharp
 {
     public sealed class GitLoaderContext : AssemblyLoadContext
     {
-        private readonly string[] assemblies;
+        private readonly Assembly entryPointAssembly;
         public static GitLoaderContext Instance { get; private set; }
 
-        private GitLoaderContext(string[] assemblies) => this.assemblies = assemblies;
+        private GitLoaderContext(Assembly entryPointAssembly) => this.entryPointAssembly = entryPointAssembly;
 
-        public static void Init(params string[] assemblies) => Instance = new GitLoaderContext(assemblies);
+        public static void Init(Assembly entryPointAssembly) => Instance = new GitLoaderContext(entryPointAssembly);
 
         protected override Assembly Load(AssemblyName assemblyName)
         {
-            if (assemblies.Contains(assemblyName.Name))
+            if (assemblyName.Name == entryPointAssembly.GetName().Name)
+            {
+                return entryPointAssembly;
+            }
+
+            var path = Path.Combine(Path.GetDirectoryName(typeof(GitLoaderContext).Assembly.Location), assemblyName.Name + ".dll");
+
+            if (File.Exists(path))
             {
-                var path = Path.Combine(Path.GetDirectoryName(typeof(GitLoaderContext).Assembly.Location), assemblyName.Name + ".dll");
                 return LoadFromAssemblyPath(path);
             }
 
-            return Default.LoadFromAssemblyName(assemblyName);
+            return null;
```

Comment:
It presumably would be messy to fully turn on nullable, since it would require modifying more code, but it would be nice to at least temporarily turn on nullable here, since you might return a null value. Then you could make sure every code path can handle null.